### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by adranwit**
- Major refactoring/restructuring
- Added cross partition sync (with removal)
- Added diff.NewIDOnly strategy
- Added DirectAppend option (to bypass transient table for inserts/streaming)

### CODE REVIEW

File: sync/app/app.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package main
 
 import (
 	"dbsync/sync"
+	"dbsync/sync/shared"
```
Comment (obalunenko) [inline sync/app/app.go:5]: Use `[gogroup](https://github.com/Bubblyworld/gogroup)` to sort and group imports

File: sync/app/app.go [hunk 2]
```diff
@@ -15,7 +16,8 @@ import (
 	"os"
 )
 
-const Version = "0.8.7"
+//Version app version
+const Version = "0.9.0"
```
Comment (obalunenko) [inline sync/app/app.go:20]: Prefer to use go build ldflags to set version variables
https://stackoverflow.com/questions/47509272/how-to-set-package-variable-using-ldflags-x-in-golang-build

File: sync/chunk/service.go [hunk 3]
```diff
@@ -0,0 +1,212 @@
+package chunk
+
+import (
+	"dbsync/sync/contract"
+	"dbsync/sync/core"
+	"dbsync/sync/criteria"
+	"dbsync/sync/dao"
```
Comment (obalunenko) [inline sync/chunk/service.go:7]: The same about sorting imports

File: sync/chunk/service.go [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+	"dbsync/sync/shared"
+	"dbsync/sync/transfer"
+	"fmt"
+)
+
+//Service represents a chunk service
+type Service interface {
+	Build(tx *shared.Context) error
+	Sync(tx *shared.Context) error
+
```
Comment (obalunenko) [inline sync/chunk/service.go:22]: A redundant empty line.

File: sync/chunk/service.go [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+	"dbsync/sync/contract/strategy"
+	"dbsync/sync/shared"
+	"dbsync/sync/transfer"
+	"fmt"
+)
+
+//Service represents a chunk service
+type Service interface {
+	Build(tx *shared.Context) error
```
Comment (obalunenko) [inline sync/chunk/service.go:20]: Add comments to interface methods

File: sync/chunk/service.go [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+	source, dest, err := s.Fetch(ctx, filter)
+	if err != nil {
+		return nil, err
+	}
+	err = s.UpdateStatus(ctx, status, source, dest, filter, s.IsOptimized())
+	return core.NewChunk(status, filter), err
+}
+
+//New creates a nwe chunk service
+func New(sync *contract.Sync, partition *core.Partition, dao dao.Service, mutex *shared.Mutex, jobService jobs.Service, transferService transfer.Service) Service {
```
Comment (obalunenko) [inline sync/chunk/service.go:202]: Add line break

File: sync/chunk/service_test.go [hunk 7]
```diff
@@ -0,0 +1,164 @@
+package chunk
+
+import (
+	"dbsync/sync/contract"
+	"dbsync/sync/core"
+	"dbsync/sync/dao"
+	"dbsync/sync/jobs"
+
+	"dbsync/sync/contract/strategy/pseudo"
```
Comment (obalunenko) [inline sync/chunk/service_test.go:9]: the same about sorting imports

File: sync/chunk/service_test.go [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+	"github.com/stretchr/testify/assert"
+	"github.com/viant/assertly"
+	"github.com/viant/dsc"
+	"github.com/viant/dsunit"
+	"github.com/viant/toolbox"
+	"path"
+	"testing"
+)
+
+var chunkerTestConfig *dsc.Config
```
Comment (obalunenko) [inline sync/chunk/service_test.go:23]: Could be merged in one block
```
var(
)
```

File: sync/chunk/service_test.go [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+var jobService jobs.Service
+
+func init() {
```
Comment (obalunenko) [inline sync/chunk/service_test.go:26]: Prefer not to use init() functions

**Comment by obalunenko** [comment]
Also, it always nice to add context to returned errors, especially when there is more than one error-case exit from the function.
https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully

File: sync/chunk/service.go (hunk 4 above)
Comment (adranwit) [reply sync/chunk/service.go:22]: Added interface comments, 
With respect to imports, I've been using
 gofmt  -s -w (I hope this should address it)

File: sync/app/app.go (hunk 2 above)
Comment (adranwit) [reply sync/app/app.go:20]: This makes entirely sense

File: sync/chunk/service_test.go (hunk 8 above)
Comment (adranwit) [reply sync/chunk/service_test.go:23]: merged

File: sync/chunk/service_test.go (hunk 9 above)
Comment (adranwit) [reply sync/chunk/service_test.go:26]: would that init be fired outside of test?

Comment (adranwit) [reply sync/chunk/service_test.go]: Then it comes to error to context, I was thinking account adding error, 
all that said I've decided that if error occurs on one partition, I would not like to terminate to other transfer, sync for the same job, altough the job end with error status.  Partitioned sync is sort of like individual table sync.  Will think it thgrough

File: sync/app/app.go (hunk 2 above)
Comment (adranwit) [reply sync/app/app.go:20]: All that said, I'd like only one place storing version, and then this to be dynamically pass to docker build (cgo dep) and e2e build,

File: sync/chunk/service_test.go (hunk 9 above)
Comment (adranwit) [reply sync/chunk/service_test.go:26]: btw I like errors.Wrap(),

**Pull request merged**